src/game/common/bitflags.h

```diff
@@ -216,20 +217,216 @@ enum ModelConditionFlagType
 template<int bits> class BitFlags
 {
 public:
+    enum BogusInitType
+    {
+        kInit = 0,
+    };
+
+    BitFlags() {}
+    BitFlags(BogusInitType type, int flag) { m_bits.set(flag); }
+    BitFlags(BogusInitType type, int flag1, int flag2)
+    {
+        m_bits.set(flag1);
+        m_bits.set(flag2);
+    }
+
+    BitFlags(BogusInitType type, int flag1, int flag2, int flag3)
+    {
+        m_bits.set(flag1);
+        m_bits.set(flag2);
+        m_bits.set(flag3);
+    }
+
+    BitFlags(BogusInitType type, int flag1, int flag2, int flag3, int flag4)
+    {
+        m_bits.set(flag1);
+        m_bits.set(flag2);
+        m_bits.set(flag3);
+        m_bits.set(flag4);
+    }
+
+    BitFlags(BogusInitType type, int flag1, int flag2, int flag3, int flag4, int flag5)
+    {
+        m_bits.set(flag1);
+        m_bits.set(flag2);
+        m_bits.set(flag3);
+        m_bits.set(flag4);
+        m_bits.set(flag5);
+    }
+
     bool operator==(BitFlags &that) const { return m_bits == that.m_bits; }
     bool operator!=(BitFlags &that) const { return m_bits != that.m_bits; }
 
     void Clear() { m_bits.reset(); }
-    void Set(unsigned bit) { m_bits.set(bit); }
-    bool Get(unsigned bit) const { return m_bits.test(bit); }
+    void Set(int bit, int value) { m_bits.set(bit, value); }
```

Comment:
are these (set, test,)  actually signed std::bitset uses std::size_t which is unsigned. I would have thought that it would use `uint32_t`.

Reply:
The available information says that yes in fact they take int.